File: bench/elixir/loop.exs
defmodule B do
  def loop(i, n, acc) when i >= n, do: acc
  def loop(i, n, acc), do: loop(i + 1, n, acc + i)
end

n = String.to_integer(System.get_env("BENCH_N") || "30000000")
IO.puts(B.loop(0, n, 0))

File: bench/erlang/loop.erl
#!/usr/bin/env escript
%%! -mode(compile)
main(_) -> io:format("~w~n", [loop(0, bench_n(30000000), 0)]).
bench_n(D) -> case os:getenv("BENCH_N") of false -> D; V -> list_to_integer(V) end.
loop(I, N, Acc) when I >= N -> Acc;
loop(I, N, Acc) -> loop(I+1, N, Acc+I).
